src/snowflake/connector/connection.py

```diff
@@ -1103,9 +1107,16 @@ def _set_parameters(self, ret, session_parameters):
                 else:
                     TelemetryService.get_instance().disable()
             elif PARAMETER_CLIENT_SESSION_KEEP_ALIVE == name:
-                self.client_session_keep_alive = value
+                # Only set if the local config is None.
+                # Always give preference to user config.
+                if self.client_session_keep_alive is None:
+                    self.client_session_keep_alive = value
+                else:
+                    client_session_keep_alive_set = True
             elif PARAMETER_CLIENT_SESSION_KEEP_ALIVE_HEARTBEAT_FREQUENCY == \
-                    name:
+                    name and self.client_session_keep_alive is True\
+                    and client_session_keep_alive_set is False:
```

Comment:
What happens if this parameter is iterated through first and then the keepalive? Should we force this parameter to be set after client keepalive?